### ISSUE TITLE
Make threshold and tolerance configurable

### ISSUE THREAD

**Issue by marcusand** [chrisguttandin/media-sync#3]
Since adjusting playbackRate and setting currentTime is costly in terms of performance and probably varies depending on the use case, it would be nice if one could set the threshold for the update strategy and the time tolerance.

**Comment by chrisguttandin** [comment]
It is now possible to configure the `threshold`, `timeConstant` and `tolerance`. The following snippet would recreate the default configuration.

```js
import {
    createSetTimingsrc,
    createUpdateGradually,
    createUpdateStepwise,
    setTimingsrcWithCustomUpdateFunction
} from 'timingsrc';

const THRESHOLD = 1;
const TIME_CONSTANT = 0.5;
const TOLERANCE = 0.025;

const customSetTimingsrc = createSetTimingsrc(
    setTimingsrcWithCustomUpdateFunction,
    window.navigator.userAgent.includes('Safari') && !window.navigator.userAgent.includes('Chrome')
        ? createUpdateStepwise(TOLERANCE)
        : createUpdateGradually(TIME_CONSTANT, THRESHOLD, TOLERANCE)
);
```

The update is published as v1.1.0.

**Comment by marcusand** [comment]
Nice, ty :)
